src/dbnode/persist/fs/types.go

```diff
@@ -576,6 +582,18 @@ type Merger interface {
 		nsCtx namespace.Context,
 		onFlush persist.OnFlushSeries,
 	) (persist.DataCloser, error)
+
+	// MergeAndCleanup merges the specified fileset file with a merge target and removes the previous version of the
+	// fileset. This should only be called within the bootstrapper. Any other file deletions outside of the bootstrapper
+	// should be handled by the CleanupManager.
```

Comment:
We could add an `isBootstrapped` bool and error if it's set to true? Migration tasks can just pass in false since they're only executed during bootstrap. Would at least make callers stop and think a bit more about the usage.